File: tests/Contexts/Rutine/Exercises/domain/Exercise.test.ts
import { ExerciseBar } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseBar';
import { ExerciseId } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseId';
import { ExerciseMuscle } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseMuscle';
import { ExerciseName } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseName';
import { ExerciseMother } from './ExerciseMother';

describe('Exercise', () => {
  it('should return a new exercise instance', () => {
    const id = 'PBTR';
    const name = 'Pess Banca Tríceps';
    const muscle = 'Tríceps';
    const bar = 10;

    const exercise = ExerciseMother.create(
      new ExerciseId(id),
      new ExerciseName(name),
      new ExerciseMuscle(muscle),
      new ExerciseBar(bar)
    );

    expect(exercise.id.value).toBe(id);
    expect(exercise.name.value).toBe(name);
    expect(exercise.muscle.value).toBe(muscle);
    expect(exercise.bar.value).toBe(bar);
  });
});

File: src/test/java/com/kuro/rutineweb/exercises/entities/ExerciseTest.java
package com.kuro.rutineweb.exercises.entities;

import com.kuro.rutineweb.Shared.entities.UserId;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.UUID;


@SpringBootTest
public class ExerciseTest {
    @Test
    public void should_return_a_new_exercise_instance() {
        UserId userId = new UserId(UUID.randomUUID().toString());
        ExerciseId id = new ExerciseId("PTR");
        ExerciseName name = new ExerciseName("Press Fatal");
        ExerciseMuscle muscle = new ExerciseMuscle("Pectoral");
        ExerciseBar bar = new ExerciseBar(10);
        ExerciseRm rm = new ExerciseRm(0);

        Exercise created = ExerciseMother.create(userId, id, name, muscle, bar, rm);

        Assertions.assertEquals(id.getValue(), created.getId().getValue());
        Assertions.assertEquals(name.getValue(), created.getName().getValue());
        Assertions.assertEquals(muscle.getValue(), created.getMuscle().getValue());
        Assertions.assertEquals(bar.getValue(), created.getBar().getValue());
    }
}
